highway.tex

```diff
@@ -615,9 +630,207 @@ \subsection{Checkpoints}
+
+We will often think of endorsed votes as being certified by the validators as a whole. While the justifications of other messages might be considered suspect (in that they might contain equivocation bombs), endorsed votes are considered to be more solid.
+\begin{definition}
+A vote $\mu$ is said to cite another vote $\lambda$ \emph{naively}, denoted $\mu \geq_n \lambda$ if $\mu \geq \lambda$ and there is no endorsed vote $\nu$ so that $\mu > \nu \geq \lambda$.
+\end{definition}
+
+We now introduce a new validity criterion:
+
+\paragraph{Limited Naivety Criterion:} A vote $\mu$ sent by validator $V$ is considered incorrect if there is an equivocation $(\lambda_1, \lambda_2)$ and two messages $\mu_1, \mu_2$ (possibly equal, possibly coinciding with $\mu$) by $V$ such that $\mu \geq \mu_1 \geq_n \lambda_1 and \mu \geq \mu_2 \geq_n \lambda_2$.
+
+Perhaps the most important property here is that this is sufficient to show that not too many equivocating votes are contained in any message justification.
+% 
+More specifically, the following theorem states that if we additionally bound the round exponent, we can ensure that no message chain is too long, and thus this will limit the number of messages that ever need to be processed during one era.
+
+\begin{theorem}\label{thm:limit_spam}
+Let $k_{min}$ be the minimum round exponent, and $2^K$ be the length of an era. Additionally, let $n$ be the total number of validators among which there are $d$ dishonest ones with cumulative weight strictly less than $\ww(\mathcal{V})/2$. Then, any given message $\mu$ created by an honest validator justifies at most $2^{K-k_{min}}\cdot (d\cdot n + n-d)$ messages.
+\end{theorem}
+\noindent 
+The following subsection is devoted to proving Theorem~\ref{thm:limit_spam}.
+
+
+%\begin{proposition}\label{chainBoundProp}
+%Consider a protocol state where all votes satisfy Criteria 1. Assume furthermore that no pair of equivocating votes are endorsed. Let $n$ be the total number of validators. Then for any message $\mu$ and any validator $W$, the set of messages by $W$ in the justification of $\mu$ are contained in the union of at most $n$ chains of messages.
+%\end{proposition}
+
+\subsubsection{Bounding the Amount of Spam}
+
+\begin{lemma}\label{chainBoundLemma}
+Consider a protocol state $\Pi$ in which all votes are valid according to Limited Naivety Criterion. Let $n$ be the total number of validators among which dishonest ones have cumulative weight strictly less than $\ww(\mathcal{V})/2$. Then for any message $\mu$ and any validator $W$, the set of messages by $W$ in the justification of $\mu$ are contained in the union of at most $n$ chains of messages.
+\end{lemma}
+
+\begin{proof}
+As there are less than $\ww(\mathcal{V})/2$ dishonest validators and no honest validator will ever endorse an equivocation, no pair of equivocating votes are endorsed in $\Pi$. 
```

Comment:
I don't think that's right; see my comment about the maximum number of endorsed equivocations above.

Reply:
Unfortunatelly you're right. :/ In general, we'll try to redefine types of validators into: honest, crashing and byzantine, bo until we all agree that it's a good idea, I'm chainging it to 2V/3, as it was (and should be).